### ISSUE TITLE
Optional attribute in class diagram

### ISSUE THREAD

**Issue by Li-Zizhen** [nus-cs2103-AY2324S2/forum#1058]
Why is C wrong? The attribute in `chair` is optional, it can choose not to show its attribute `table`.

![image](https://github.com/nus-cs2103-AY2324S2/forum/assets/122284239/829d9f41-4b7d-4f38-bf34-88385d5ad624)

**Comment by jannaleong** [comment]
in the class diagram, there is no association between chair and table class. If they had left out attribute in chair, there needs to be a line between the 2 classes.

As the object diagram shows there is an associaiton, this class diagram in c is not compliant.

**Comment by Li-Zizhen** [comment]
@jannaleong
Thank you for your response. So if we omit the attribute, we much show the association line instead?
 ![image](https://github.com/nus-cs2103-AY2324S2/forum/assets/122284239/d173fe5a-9dcb-46d2-b36b-4db4f9303418)

**Comment by drustanyjt** [comment]
I don't think that's the main focus. In the Class Diagram you highlighted, there is no association line being drawn between Chair and Table, hence there should be no associations between Chair and Table. This means that the object diagram is incorrect (because it IS showing an association between the Chair and Table)

**Comment by Li-Zizhen** [comment]
Sorry for that. 

Showing attributes is the same as the association line. Therefore, we can use attributes instead of association lines. However, attributes can also be omitted. 

Therefore, if we use attributes instead of the association line and then omit that attribute(consider it as not important), we can just not show it right?

**Comment by ThatGirlJam** [comment]
I suppose that's true, but since this question seems to be specific to the object diagram given, where it was deemed important to have that association between Chair and Table, c is less likely to be a correct option.

**Comment by damithc** [comment]
> I suppose that's true, but since this question seems to be specific to the object diagram given, where it was deemed important to have that association between Chair and Table, c is less likely to be a correct option.

Yup, if it was relevant to the answer, it wouldn't have been omitted as 'not important'.
